src/server/conn.rs

```diff
@@ -160,18 +166,33 @@ impl Conn {
             self.res.pop_front();
         }
 
-        // no data for writing, remove writable.
+        // no data for writing, remove writable
         self.interest.remove(EventSet::writable());
-        self.reregister(event_loop)
+        try!(self.reregister(event_loop));
+
+        self.on_write_complete(true);
```

Comment:
no guarantee here. this callback just means being called when writing all data ok. 

Refer tornado 3.2 https://github.com/tornadoweb/tornado/blob/branch3.2/tornado/iostream.py